dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,176 @@
+import math
+import numpy as np
+from sklearn.base import clone
+from sklearn.model_selection import ParameterSampler
+
+from toolz import groupby
+from distributed import as_completed, default_client
+from dask_searchcv.model_selection import DaskBaseSearchCV
+
+
+def _get_nr(R, eta=3):
+    s_max = math.floor(math.log(R, eta))
+    B = (s_max + 1) * R
+
+    brackets = list(reversed(range(int(s_max + 1))))
+    N = [math.ceil(B / R * eta**s / (s + 1)) for s in brackets]
+    R = [int(R * eta**-s) for s in brackets]
+    return list(map(int, N)), list(map(int, R)), brackets
+
+
+def _partial_fit(model_and_meta, x, y, meta=None, fit_params={},
+                 dry_run=False):
+    model, m = model_and_meta
+    if meta is None:
+        meta = m
+    while meta['iterations'] < meta['partial_fit_calls']:
+        if not dry_run:
+            model.partial_fit(x, y, **fit_params)
+        meta["iterations"] += 1
+    return model, meta
+
+
+def _score(model_and_meta, x, y, seed=42, dry_run=False):
+    seed = seed % 2**32 - 1
+    model, meta = model_and_meta
+    rng = np.random.RandomState(seed)
+    if dry_run:
+        score = rng.rand()
+    else:
+        # TODO: change this to dask_ml.get_scorer
+        #  score = model.score(x, y)
+        score = rng.rand()
+    meta.update(score=score)
+    return meta
+
+
+def _top_k(results, k=1, sort="score"):
```

Comment:
Is this equivalent to 

```python
sorted(results, key=lambda x: x['score'])[:k]
```

And just to make sure, are higher numbers better here? With sklearn, whenever the term "score" is used it's assumed that higher is better.

Reply:
Nice. And correct, higher numbers are better.

Reply:
If possible I prefer that we avoid creating tiny functions like this.  It often helps with the writing process but can make it difficult when reviewing or reading code when debugging.  Readers have to dereference things frequently.   If we can just use the `sorted` or `topk` functions inline and achieve the same effect then I suggest that we do that.